e2e/ansible/playbooks/feature/snapshots/kubectl_snapshot/test-kubectl-snapshot.yml

```diff
@@ -0,0 +1,297 @@
+
+        - name: 7) Creating the volume snapshot
+          shell: source ~/.profile; kubectl create -f "{{ snapshot }}"
+          args:
+            executable: /bin/bash
+          register: sp
+          delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+          until: '"volumesnapshot \"snapshot-demo\" created" in sp.stdout'
+          delay: 60
+          retries: 5
+
+        - name: 7a) Check if the snapshot is created successfully
+          shell: sleep 20 ; source ~/.profile; kubectl describe volumesnapshot snapshot-demo | grep Message
+          args:
+            executable: /bin/bash
+          register: sp_out
+          delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+          until: "'Snapshot created successfully' in sp_out.stdout"
+          delay: 60
+          retries: 5
+
+        - name: 8) Creating the snapshot-promoter storage class
+          shell: source ~/.profile; kubectl create -f "{{ snapshot_sc }}"
+          args:
+            executable: /bin/bash
+          register: sc
+          delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+          until: "'storageclass \"snapshot-promoter\" created' in sc.stdout"
+          delay: 30
+          retries: 5
+
+        - name: 9) Creating PVC from the snapshot
+          shell: source ~/.profile; kubectl create -f "{{ snapshot_claim }}"
+          args:
+            executable: /bin/bash
+          register: claim_out
+          delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+          until: "'persistentvolumeclaim \"demo-snap-vol-claim\" created' in claim_out.stdout"
+          delay: 60
+          retries: 5
+
+        - name: 9a) Checking if the pvc is created successfully
+          shell: source ~/.profile; kubectl get pvc | grep demo-snap-vol-claim
+          args:
+            executable: /bin/bash
+          register: pvc_out
+          delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+          until: "'Bound' in pvc_out.stdout"
+          delay: 60
+          retries: 10
+
+        - name: 10) Obtaining new PV name
+          shell: source ~/.profile; kubectl get pv | grep demo-snap-vol-claim | awk {'print $1'}
+          args:
+            executable: /bin/bash
+          register: snap_pv
+          delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+
+        - name: 10a) Obtaining the new controller pod name
+          shell: source ~/.profile; kubectl get po | grep {{ snap_pv.stdout }} | grep ctrl | awk {'print $1'}
+          args:
+            executable: /bin/bash
+          register: new_ctrl_pod
+          delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+
+        - name: 10b) Checking if the sync completed
+          shell: source ~/.profile; kubectl logs {{new_ctrl_pod.stdout}}
+          args:
+            executable: /bin/bash
+          register: ctrl_log
+          delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+          until: "'Done synchronizing' in ctrl_log.stdout"
```

Comment:
@gprasath , have we investigated if this string 'Done synchronizing' reused to indicate any sync process other than the clone ?

Reply:
Yes @ksatchit . In the controller log, this string has been used only after completing the synchronization.